### PR TITLE
Add fine-tuning task 

### LINKED ISSUE #23: Add fine-tuning task (based on a pre-trained model)

### PR DESCRIPTION
**Pull request by KennethEnevoldsen**
This pull request adds tests for the fine-tuning task to ensure the accuracy and reliability of the model as well as the fine-tuning task itself. 

- Working on implementing the classification module:
    - Added tests for intended functionality 
    - Added filtering for individuals without any of the specified phenotypes
      - This approach allows predicting multiple phenotypes at once (it is also the easiest to implement)
    - Added the masking strategy for masking out all of the specified phenotypes
      - Noted a bug pertaining to the masking of phenotypes in the original pre-training  fixed it
      - Added a test to make sure that that issue doesn’t happen again
- Implemented forward pass through the model and loss calculation
  - Added test to see if it backprops correctly (there is still a few test not passing here, but will hopefully have some time to finish that tomorrow)
  - Added logging for accuracy (for all and pr. Phenotype)
- Need to:
  - Add configs
    - these are kinda recipes which specified everything (which is then logged for reproducibility). You can see one here: 
      https://github.com/KennethEnevoldsen/snp-transformer/blob/main/src/projects/pretrain/config.cfg
  - Test that it can load from a pre-trained model.
  - Test that it works with the trainer using a GPU (both with and without continual training)
  - Test that logging works and logs all the required metrics 

fixes #23

### CODE REVIEW

**Comment by KennethEnevoldsen** [comment]
- Continued working on implementing the classification module:
  - Had some issues with a matrix manipulation function that didn't work as expected (fixed but took a while) - Apparently `matrix[mask][i] = 1` does not change `matrix`.
  - Added config for testing classification works with trainer
  - Added config for testing if classification can be loaded from checkpoint
  
- Ongoing:
  - Test that it works with the trainer using a GPU (both with and without continual training)
  - Test that logging works and logs all the required metrics
